### ISSUE TITLE
initialization crash

### ISSUE THREAD

**Issue by benibela** [ACRA/acra#780]
I have found a crash report in the app store which appears to come from the ACRA initialization:

* java.util.ServiceConfigurationError: 
 
  at java.util.ServiceLoader.fail (ServiceLoader.java:233)
 
  at java.util.ServiceLoader.parse (ServiceLoader.java:310)
 
  at java.util.ServiceLoader.-wrap0 (Unknown Source)
 
  at java.util.ServiceLoader$LazyIterator.hasNextService (ServiceLoader.java:358)
 
  at java.util.ServiceLoader$LazyIterator.hasNext (ServiceLoader.java:402)
 
  at java.util.ServiceLoader$1.hasNext (ServiceLoader.java:488)
 
  **at f.a.q.f.a (ServicePluginLoader.java:6)**
 
  **at f.a.i.c.a (BaseCoreConfigurationBuilder.java:3)**
 
  **at [de.benibela.videlibri.VideLibriApp.onCreate (VideLibriApp.kt:7)](https://github.com/benibela/videlibri/blob/master/android/android/src/de/benibela/videlibri/VideLibriApp.kt#L35)**
 
  at android.app.Instrumentation.callApplicationOnCreate (Instrumentation.java:1125)
 
  at android.app.ActivityThread.handleBindApplication (ActivityThread.java:6062)
 
  at android.app.ActivityThread.-wrap1 (Unknown Source)
 
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:1764)
 
  at android.os.Handler.dispatchMessage (Handler.java:105)
 
  at android.os.Looper.loop (Looper.java:164)
 
  at android.app.ActivityThread.main (ActivityThread.java:6944)
 
  at java.lang.reflect.Method.invoke (Native Method)
 
  at com.android.internal.os.Zygote$MethodAndArgsCaller.run (Zygote.java:327)
 
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1374)

* Caused by: java.net.MalformedURLException: 
 
  at java.net.URL.<init> (URL.java:624)
 
  at java.net.URL.<init> (URL.java:487)
 
  at java.net.URL.<init> (URL.java:436)
 
  at java.net.JarURLConnection.parseSpecs (JarURLConnection.java:175)
 
  at java.net.JarURLConnection.<init> (JarURLConnection.java:158)
 
  at libcore.io.ClassPathURLStreamHandler$ClassPathURLConnection.<init> (ClassPathURLStreamHandler.java:136)
 
  at libcore.io.ClassPathURLStreamHandler.openConnection (ClassPathURLStreamHandler.java:84)
 
  at java.net.URL.openConnection (URL.java:992)
 
  at java.net.URL.openStream (URL.java:1058)
 
  at java.util.ServiceLoader.parse (ServiceLoader.java:305)

* Caused by: java.lang.StringIndexOutOfBoundsException: 
 
  at java.lang.String.substring (String.java:1975)
 
  at java.net.URLStreamHandler.parseURL (URLStreamHandler.java:285)
 
  at sun.net.www.protocol.file.Handler.parseURL (Handler.java:67)
 
  at java.net.URL.<init> (URL.java:619)

with acraVersion 5.5.0 on  Samsung Galaxy S7 (herolte), 3584MB RAM, Android 8.0.

**Comment by F43nd1r** [comment]
This looks like a broken classloader. Are you using any special class loading in your project?

**Comment by benibela** [comment]
I am not doing anything special.

I use Kotlin and OkHttp. I do not know if they are doing special things.

When I added OkHttp I had the problem that the default class loader of JNI, env.findClass(env), could not find OkHttp.  I had to keep a reference to the classloader used during onCreate and then use its find method.  But that should  not matter here, since I only load the JNI so after initializing Acra. 

Perhaps it is just a broken phone. The App Store reports are always like this.
